mLRS/Common/hal/esp/tx-hal-radiomaster-tx16s-2400-esp32.h

```diff
@@ -0,0 +1,214 @@
+//*******************************************************
+// Copyright (c) MLRS project
+// GPL3
+// https://www.gnu.org/licenses/gpl-3.0.de.html
+//*******************************************************
+// hal
+//*******************************************************
+
+/*
+  Flashing ESP8285 backpack WiFi bridge:
+  - Board: Generic ESP8266 Module, define LED_IO 16
+  - First flash can be via web browser/ELRS WiFi
+  - Flash backpack before erasing ELRS or flash ELRS first using configurator via EdgeTx passthrough
+  - Need ELRS version of esptool from https://github.com/ExpressLRS/Backpack
+  - Power up radio, plug in USB, select VCP
+  - On Linux, run something like "Backpack/python/external/esptool/esptool.py --passthrough --port /dev/ttyACM0 --baud 460800 --before etx --after hard_reset write_flash 0x0000 ~/Arduino/build/mlrs-wireless-bridge-esp8266.ino.bin
+
+  Flashing ESP32 module:
+  - Need ELRS python folder from https://github.com/ExpressLRS/ExpressLRS
+  - Power up radio, plug in USB, select VCP
+  - On Linux, run something like "python ExpressLRS/src/python/ETXinitPassthrough.py"
+  - Use Visual Studio Code or esptool (or maybe web flasher?) to flash via ACM serial port.
+*/
+
+//-------------------------------------------------------
+// ESP32, Radiomaster TX16S internal ELRS
+//-------------------------------------------------------
+/*
+    Pin Defs
+    "serial_rx": 3,
+    "serial_tx": 1,
+    "radio_busy": 21,
+    "radio_dio1": 4,
+    "radio_dio2": 22,
+    "radio_miso": 19,
+    "radio_mosi": 23,
+    "radio_nss": 5,
+    "radio_rst": 14,
+    "radio_sck": 18,
+    "radio_dcdc": true,
+    "power_rxen": 27,
+    "power_txen": 26,
+    "power_lna_gain": 12,
+    "power_min": 0,
+    "power_high": 4,
+    "power_max": 4,
+    "power_default": 2,
+    "power_pdet": 35,
+    "power_pdet_intercept": 1.5,
+    "power_pdet_slope": 0.032,
+    "power_control": 0,
+    "power_values": [-17,-13,-9,-6,-2],
+    "use_backpack": true,
+    "debug_backpack_baud": 460800,
+    "debug_backpack_rx": 16,
+    "debug_backpack_tx": 17,
+    "backpack_boot": 15,
+    "backpack_en": 25
+*/
+
+
+#define DEVICE_HAS_JRPIN5
+#define JRPIN5_INTERNAL // Full duplex CRSF interface for Internal modules
```

Comment:
usually we only have DEVICE_HAS_xxx at the top

(and these usually can be changed without the things breaking apart). A specifier like JRPIN5_INTERNAL should go somewhere below.

Reply:
I haven't used this yet anywhere.  But we will probably need something like this eventually.

Reply:
then please at least leave a 

#ifndef JRPIN5_INTERNAL #error #endif in the esp jrpin5 file

Reply:
I think I'll just remove this define for now.